Ut p4

vishala-r opened the pull request:
This patch implements UT for p4 package. 
This also includes PR: https://github.com/ipdk-io/k8s-infra-offload-old/pull/67 raised on old k8s repo

dandaly reviewed (approved):
Looks like a good start here.

ahalimx86 committed:
handle errors from health check watcher error channel

This fixes the coverity scan issue on error checking on error channel
from health check watcher.

Signed-off-by: Abdul Halim <abdul.halim@intel.com>

ahalimx86 committed:
fix utils unit test for watcher functions

fix unit tests for watcher functions as it seemed the watcher would wait
forever to receive values for error channel.

Signed-off-by: Abdul Halim <abdul.halim@intel.com>

Commit:
Upgrade to Go 1.19 version and update http2 package for fix snyk reported vulnerabilities

Signed-off-by: Tripathi, Dhruv Prakash <dhruv.tripathi@intel.com>

Commit:
Updated entire package to Go 1.19.4

Signed-off-by: Tripathi, Dhruv Prakash <dhruv.tripathi@intel.com>

mgaravin committed:
set workflow write permission to packages

Signed-off-by: Aravindkumar Mg <aravindkumar.mg@intel.com>

mgaravin committed:
bump go version to 1.19.4

Signed-off-by: Aravindkumar Mg <aravindkumar.mg@intel.com>

mgaravin committed:
disabled unit test based on team request

Signed-off-by: Aravindkumar Mg <aravindkumar.mg@intel.com>

ipdksw committed:
Merge pull request #1 from ipdk-io/fix_docker_push

set workflow write permission to packages

vishala-r committed:
Implemented UT for p4 package

Signed-off-by: Vishalakshi <vishalakshi.r@intel.com>
